src/core_ocean/mode_init/mpas_ocn_init_tidal_boundary.F

```diff
@@ -261,6 +280,58 @@ subroutine ocn_init_setup_tidal_boundary(domain, iErr)!{{{
         end do
       end if
 
+      !! zhendong@2020-03-03
+      if (config_idealized_profile) then
+        call mpas_pool_get_array(forcingPool, 'bottomDrag', bottomDrag)
+        config_idealized_profile_Lmarsh = 1.0_RKIND - config_idealized_profile_Lcoast &
+                                                    - config_idealized_profile_Lshore
+        if (config_idealized_profile_Lmarsh .lt. 0.0_RKIND) then
+          call mpas_log_write("Lshore+Lcoast cannot be bigger than 1.0")
+          iErr = 1
+        endif
+        cff1 = yMax*config_idealized_profile_Lcoast
+        cff2 = yMax*config_idealized_profile_Lmarsh
+        cff3 = yMax*config_idealized_profile_Lshore
+        ! by defining the slopes and left_bottom_depth, the pre-defined right_bottom_depth won't work.
+        ! Need redefine it.
+        config_tidal_boundary_right_bottom_depth = config_tidal_boundary_left_bottom_depth + &
+                                                   cff1*config_idealized_profile_Scoast + &
+                                                   cff2*config_idealized_profile_Smarsh + &
+                                                   cff3*config_idealized_profile_Sshore
+        do iCell = 1, nCellsSolve
+          if (yCell(iCell) .lt. cff1) then
+            bottomDepth(iCell) = config_tidal_boundary_left_bottom_depth &
+              + (yCell(iCell)-yMin)*config_idealized_profile_Scoast
+            dep_mark1 = bottomDepth(iCell)
+            bottomDrag(iCell) = config_idealized_profile_Cd
+          elseif (yCell(iCell) .lt. (cff1+cff2)) then
+            bottomDepth(iCell) = dep_mark1 + (yCell(iCell)-cff1)*config_idealized_profile_Smarsh
+            dep_mark2 = bottomDepth(iCell)
+            bottomDrag(iCell) = config_idealized_profile_Cd_marsh
+          else
+            if (cff2 .eq. 0.0_RKIND) dep_mark2 = dep_mark1
+            bottomDepth(iCell) = dep_mark2 + (yCell(iCell)-cff1-cff2)*config_idealized_profile_Sshore
+            bottomDrag(iCell) = config_idealized_profile_Cd
+          endif
+        end do !! do iCell
+      else
+      ! if not config_idealized_profile, assign a constant slope
+        do iCell = 1, nCellsSolve
+          bottomDepth(iCell) = config_tidal_boundary_left_bottom_depth &
+            + (yCell(iCell) - yMin) / (yMax - yMin)  * &
+            (config_tidal_boundary_right_bottom_depth - config_tidal_boundary_left_bottom_depth)
+        end do
+      end if   !! if config_idealized_profile
+
+      if (config_tidal_boundary_right_bottom_depth < config_tidal_boundary_left_bottom_depth) then
+        call mpas_log_write('Right boundary must be deeper than left boundary!', MPAS_LOG_CRIT)
+      end if
+
+      ! Set refBottomDepth, bottomDepth, and maxLevelCell
```

Comment:
This is inconsistent with previous code: https://github.com/MPAS-Dev/MPAS-Model/pull/552/files#diff-6270014bc9ae9b8554f8af50127caf5fR263-R267 and seems to assume `config_tidal_boundary_left_bottom_depth = 0`.  The need to modify this is to accomodate other test cases of this type that may have `config_tidal_boundary_right_bottom_depth /=0`.

```
 ! Set refBottomDepth, bottomDepth, and maxLevelCell
      do k = 1, nVertLevels
        refBottomDepth(k) = config_tidal_boundary_left_bottom_depth + &
          (config_tidal_boundary_right_bottom_depth - config_tidal_boundary_left_bottom_depth)* interfaceLocations(k+1)
      end do
```